src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -1596,24 +1611,1318 @@ struct npc_silverpine_forsaken_bat : public VehicleAI
+                break;
+            case DISPLAY_FEMALE_03_F:
+                displayId = DISPLAY_FEMALE_03_D_F;
+                break;
+            case DISPLAY_FEMALE_04_F:
+                displayId = DISPLAY_FEMALE_04_D_F;
+                break;
+            default:
+                break;
+        }
+
+        me->SetDisplayId(displayId);
+
+        _events.ScheduleEvent(EVENT_REFUGEE_JUST_RISEN, 1s);
+
+        me->DespawnOrUnsummon(4s);
+    }
+
+    void Reset() override
+    {
+        _events.Reset();
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _events.Update(diff);
+
+        while (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_REFUGEE_JUST_RISEN:
+                    Talk(TALK_REFUGEE_JUST_RISEN);
+                    break;
+                default:
+                    break;
+            }
+        }
+    }
+
+private:
+    EventMap _events;
+};
+
+enum WorgenSentry
+{
+    SPELL_BATTLE_ROAR                           = 6507,
+    SPELL_UNDYING_FRENZY                        = 80515,
+
+    EVENT_UNDYING_FRENZY                        = 1,
+
+    TALK_SENTRY_WARNING                         = 0
+};
+
+// 44987 - Worgen Sentry
+struct npc_silverpine_worgen_sentry : public ScriptedAI
+{
+    npc_silverpine_worgen_sentry(Creature* creature) : ScriptedAI(creature) {}
+
+    void Reset() override
+    {
+        _events.Reset();
+    }
+
+    void JustDied(Unit* /*killer*/) override
+    {
+        Talk(TALK_SENTRY_WARNING);
+    }
+
+    void JustEngagedWith(Unit* /*who*/) override
+    {
+        DoCastSelf(SPELL_BATTLE_ROAR);
+
+        _events.ScheduleEvent(EVENT_UNDYING_FRENZY, 3s);
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _events.Update(diff);
+
+        while (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_UNDYING_FRENZY:
+                    DoCastVictim(SPELL_UNDYING_FRENZY);
+                    _events.Repeat(10s, 12s);
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        if (!UpdateVictim())
+            return;
+
+        DoMeleeAttackIfReady();
+    }
+
+private:
+    EventMap _events;
+};
+
+// 80515 - Undying Frenzy
+class spell_silverpine_undying_frenzy : public AuraScript
+{
+    PrepareAuraScript(spell_silverpine_undying_frenzy);
+
+    bool Validate(SpellInfo const* spellInfo) override
+    {
+        return ValidateSpellInfo({ spellInfo->GetEffect(EFFECT_1).TriggerSpell });
+    }
+
+    void HandlePeriodic(AuraEffect const* /*aurEff*/)
+    {
+        GetTarget()->CastSpell(nullptr, GetSpellInfo()->GetEffect(EFFECT_1).TriggerSpell, true);
+    }
+
+    void Register() override
+    {
+        OnEffectPeriodic += AuraEffectPeriodicFn(spell_silverpine_undying_frenzy::HandlePeriodic, EFFECT_1, SPELL_AURA_PERIODIC_TRIGGER_SPELL);
```

Comment:
this should work out of the box without script because `SPELL_AURA_PERIODIC_TRIGGER_SPELL` is exactly doing this?

Reply:
Feedback applied.